File: API/Models/Product/Review.cs
using System.ComponentModel.DataAnnotations;

namespace API.Models.ProductModel
{
    public class Review
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(500)]
        public string Pros { get; set; }

        [Required]
        [MaxLength(500)]    
        public string Cons { get; set; }

        public string Comment { get; set; }
        public DateTime CreatedAt { get; set; }

        [Required]
        [Range(10, 50)]
        public int RatingScore { get; set; }

        [MaxLength(255)]
        public string OrderStatus { get; set; }

        [Required]
        public string UserId { get; set; }
        public ApplicationUser User { get; set; }
    
        [Required]
        public int ProductId { get; set; }
        public Product Product { get; set; }
    }
}

File: client/src/app/_models/Review.ts
export interface Review {
  id: number;
  pros: string;
  cons: string;
  comment: string;
  ratingScore: number;
  orderStatus: string;
  member: Member;
  createdAt: Date;
  productId: number;
}

export interface Member {
  id: string;
  email: string;
  firstName: string;
  lastName: string;
}

export interface ReviewResponse {
  items: Review[],
  page: number,
  pageSize: number,
  totalCount: number
}

export interface ReviewRequest {
  pros: string,
  cons: string,
  comment: string,
  ratingScore: number;
  orderStatus?: string;
  productId: number;
}